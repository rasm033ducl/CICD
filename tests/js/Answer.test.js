import React from 'react';

import TestRenderer from 'react-test-renderer';

import Testee from '../../src/js/Answer';

describe( '<Answer />', () => {
	test( 'should render an answer according to the passed props', () => {
		const id = 'some-id-here';
		const text = 'Some text here.';
		const testRenderer = TestRenderer.create(
			<Testee id={ id } text={ text } />
		);

		const output = testRenderer.toJSON();

		expect( output.type ).toBe( 'li' );
		expect( output.props.className ).toContain( 'oracle__answer' );
		expect( output.children ).toContain( id );

		expect( output ).toMatchSnapshot();
	} );
} );
