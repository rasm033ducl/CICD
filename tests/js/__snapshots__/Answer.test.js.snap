// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Answer /> should render an answer according to the passed props 1`] = `
<li
  className="oracle__answer"
>
  Answer 
  some-id-here
  : 
  <em>
    Some text here.
  </em>
</li>
`;
